shared/simulator/src/bullet.rs

```diff
@@ -85,23 +85,30 @@ pub fn destroy(sim: &mut Simulation, handle: BulletHandle) {
     );
 }
 
+/// Process movement and collisions for each bullet in the simulation
+///
+/// NOTE: As a performance optimization, colliders are only added to
+/// bullets after a collision has been detected since most bullets
+/// spend most of their lifetime far away from any ships they could
+/// hit.
 pub fn tick(sim: &mut Simulation) {
     let dt = PHYSICS_TICK_LENGTH;
+    // Indices for ships
     let (indices_by_team, coarse_grids_by_team) = build_indices(sim, dt);
     let mut stack = Vec::new();
     let shape = rapier2d_f64::geometry::Ball { radius: 1.0 };
     let bullets: Vec<BulletHandle> = sim.bullets.iter().cloned().collect();
     for handle in bullets {
-        let team = {
-            let data = data_mut(sim, handle);
-            data.ttl -= dt as f32;
-            if data.ttl <= 0.0 {
-                destroy(sim, handle);
-                continue;
-            }
-            data.team
-        };
```

Comment:
I was having a little trouble parsing this when I was looking at it. It was unclear to me why mutations were being done inside the constructor for the team variable, so I restructured the code to make it a little easier to read.